### PR TITLE
Revert getPathSuffixUnescaped

### PR DESCRIPTION
**Pull request by kisunji**
Fixes https://github.com/hashicorp/consul/issues/13236
1.11.x: https://github.com/hashicorp/consul/pull/13265
1.10.x: https://github.com/hashicorp/consul/pull/13264

### Description
This PR reverts all of the changes made in https://github.com/hashicorp/consul/issues/11258 as it turns out that URLs are natively escaped and we had not tested the HTTP requests 🤦 

### Testing & Reproduction steps
```
❯ consul agent -dev
...
❯ consul services register -name test/%
Registered service: test/%
```

Before PR
```
❯ curl --request GET localhost:8500/v1/agent/service/test%2F%25
failure in unescaping path param "test/%": invalid URL escape "%"%
```

With PR
```
❯ curl --request GET localhost:8500/v1/agent/service/test%2F%25
{
    "ID": "test/%",
    "Service": "test/%",
    "Tags": [],
    "Meta": {},
    "Port": 0,
    "Address": "",
    "Weights": {
        "Passing": 1,
        "Warning": 1
    },
    "EnableTagOverride": false,
    "ContentHash": "d31fe3644baf62d8",
    "Namespace": "default",
    "Partition": "default",
    "Datacenter": "dc1"
}
```

### CODE REVIEW

File: agent/catalog_endpoint.go [hunk 1]
```diff
@@ -356,16 +357,8 @@ func (s *HTTPHandlers) catalogServiceNodes(resp http.ResponseWriter, req *http.R
 		args.TagFilter = true
 	}
 
-	if _, ok := params["merge-central-config"]; ok {
-		args.MergeCentralConfig = true
-	}
```
Comment (kyhavlov) [inline agent/catalog_endpoint.go:361]: Should this be getting reverted too? It doesn't seem related to the url escape changes

Comment (kisunji) [reply agent/catalog_endpoint.go:361]: oh good catch! Probably a victim of my deletes

File: agent/health_endpoint.go [hunk 2]
```diff
@@ -203,10 +192,6 @@ func (s *HTTPHandlers) healthServiceNodes(resp http.ResponseWriter, req *http.Re
 		args.TagFilter = true
 	}
 
-	if _, ok := params["merge-central-config"]; ok {
```
Comment (jkirschner-hashicorp) [inline agent/health_endpoint.go:206]: @kisunji : same question as https://github.com/hashicorp/consul/pull/13256/files#r882931918 - is this delete supposed to be here?

**Review by jkirschner-hashicorp — COMMENTED** [review]
@kisunji : found another place we seem to be removing the code that @kyhavlov spotted, and maybe shouldn't be removed.

**Comment by kisunji** [comment]
Thanks for the thorough review @kyhavlov and @jkirschner-hashicorp! Turns out the merge-central-config changes didn't land in enterprise when I started this PR from the enterprise side and I overwrote the files in OSS

**Comment by jkirschner-hashicorp** [comment]
@kyhavlov : looks good to me, but I defer to you (as an actual member of the eng team) to give the official approval.

**Comment by DavidOhsieAtTurbo** [comment]
Just a general comment.  Would it be easier to simply redefine getPathSuffixUnescaped to not do the double escape rather than finding all the places where that is called and removing them?

**Comment by kisunji** [comment]
> Just a general comment. Would it be easier to simply redefine getPathSuffixUnescaped to not do the double in escape rather than finding all the places where that is called and removing them?

`getPathSuffixUnescaped` was returning errors from the extra escaping, otherwise it was simply calling `strings.TrimPrefix` under the hood. I think it's cleaner to remove the abstraction (original code looked like this in the first place).

**Comment by askmike1** [comment]
Is there an ETA for this getting released (I'm assuming as part of 1.12.3)?

**Comment by jkirschner-hashicorp** [comment]
@askmike1 : yes, the current plan is for this to ship in 1.12.3. I'll keep this comment up-to-date: https://github.com/hashicorp/consul/issues/13236#issuecomment-1138893291
